internal/templates/codegen/main.go

```diff
@@ -0,0 +1,288 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+	"github.com/pkg/errors"
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/set"
+	"golang.stackrox.io/kube-linter/internal/stringutils"
+	"golang.stackrox.io/kube-linter/internal/utils"
+	"k8s.io/gengo/parser"
+	"k8s.io/gengo/types"
+)
+
+var (
+	knownNonTemplateDirs = set.NewFrozenStringSet("all", "codegen", "util")
+)
+
+const (
+	metadataMarker = "+"
+
+	paramsStructName = "Params"
+)
+
+type templateElem struct {
+	ParamDesc check.ParameterDesc
+	ParamJSON string
+}
+
+const (
+	fileTemplateStr = `// Code generated by kube-linter template codegen. DO NOT EDIT.
+// +build !templatecodegen
+
+package params
+
+import (
+	"github.com/pkg/errors"
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/templates/util"
+)
+
+var (
+	// Use this in case it doesn't get used otherwise.
+	_ = util.MustParseParameterDesc
+
+{{ range . }}
+	{{ .ParamDesc.Name}}paramDesc = util.MustParseParameterDesc({{backtick}}
```

Comment:
{{ .ParamDesc.Name}}ParamDesc? to keep camel case

Reply:
Good catch, done.